TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,15 +141,150 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
+                            };
+                        }
+                        else
+                        {
+                            IEnumerable<string> badgesFormatted = intersectList.Select( kvp => $"{kvp.Key}" );
+                            return new CommandResult
+                            {
+                                Response = $"{user.Name} is owning missing duplicates of the following badge(s): {string.Join(", ", badgesFormatted)}",
+                                ResponseTarget = ResponseTarget.WhisperIfLong
+                            };
+                        }
+                    }
+                }
+                else if( mode.Equals( PokedexModeComplementFrom ) )
+                {
+                    if( !optionalUser2.IsPresent )
+                    {
+                        return new CommandResult
+                        {
+                            Response = $"Mode {PokedexModeComplementFromDupes} requires a second user argument"
+                        };
+                    }
+                    else
+                    {
+                        ImmutableSortedDictionary<PkmnSpecies, int> myNumBadgesPerSpecies =
+                            await _badgeRepo.CountByUserPerSpecies(optionalUser2.Value.Id);
+
+                        var intersectList = numBadgesPerSpecies.Except( myNumBadgesPerSpecies );
+                        if( !intersectList.Any() )
+                        {
+                            return new CommandResult
+                            {
+                                Response = $"{user.Name} does not own any Pokémon badges {optionalUser2.Value.Name} is missing"
+                            };
+                        }
+                        else
+                        {
+                            IEnumerable<string> badgesFormatted = intersectList.Select( kvp => $"{kvp.Key}" );
+                            return new CommandResult
+                            {
+                                Response = $"{user.Name} is owning the following missing badge(s): {string.Join(", ", badgesFormatted)}",
+                                ResponseTarget = ResponseTarget.WhisperIfLong
+                            };
+                        }
+                    }
+                }
+                else if( mode.Equals( PokedexModeModes ) )
+                {
+                    return new CommandResult
+                    {
+                        Response =  $"Supported modes are \"{PokedexModeDupes}\": Show duplicate badges, \"{PokedexModeMissing}\": Show missing badges, "
+                        + "\"" + PokedexModeComplementFrom + "\": Compares missing badges from User A with owned badges from User B, "
+                        + "\"" + PokedexModeComplementFromDupes + "\": Compares missing badges from User A with owned duplicate badges from User B"
```

Comment:
- you could avoid the escaped quotes by using single quotes inside instead
- you can use string formatting for lines 2 and 3 too. looks weird that you did use it in the first but not the other lines.

but this entire this might be easier to look at using verbatim strings (`@`-prefixed) instead. you might want to check those out.

Reply:
I can NOT get myself to use verbatim strings. the code looks awful after that. But I got the escape quotes to work after a tip from you which is fine

Reply:
sure